### PR TITLE
feat: add zks_getBytecodeByHash

### LINKED ISSUE #144: zks_api: Implement `zks_getBytecodeByHash` method
## **Description:**
We need to implement the [`zks_getBytecodeByHash`](https://github.com/matter-labs/zksync-era/blob/a980ac61f484a3b6bdbe11e606977d0d92d4e9ff/core/lib/zksync_core/src/api_server/web3/backend_jsonrpc/namespaces/zks.rs#L100) method. This function fetches the bytecode of a transaction given its hash.

```rust
/// Retrieves the bytecode of a transaction based on its hash.
fn zks_getBytecodeByHash(&self, hash: H256) -> jsonrpc_core::BoxFuture<jsonrpc_core::Result<Vec<u8>>> {
    not_implemented("zks_getBytecodeByHash")
}
```

#### Expected response
Command:
```
curl -X POST -H "Content-Type: application/json" \
--data '{"jsonrpc": "2.0", "id": 1, "method": "zks_getBytecodeByHash", "params": [ "0x0100067d861e2f5717a12c3e869cfb657793b86bbb0caa05cc1421f16c5217bc" ]}' \
"https://mainnet.era.zksync.io"
```

Response:
```
{
    "jsonrpc": "2.0",
    "result": [
        0,
        4,
        0,
        0,
        0,
        0,
        0,
        2,
        0,
        11,
        ...,
        ...,
            ],
    "id": 1
}
```

### **Requirements:**

1. **Implement the Method:**
    - The method should accept a `hash` of type `H256`.
    - It should return the bytecode of the transaction as a vector of bytes (`Vec<u8>`).

2. **Unit Tests:**
    - Write unit tests ensuring the `zks_getBytecodeByHash` method functions correctly.
    - Test various edge cases, including non-existent transaction hashes.
    - Properly handle potential exceptions and errors.

3. **Inline Documentation:**
    - Write inline Rust documentation (`///` comments) explaining the purpose and usage of the `zks_getBytecodeByHash` method.
    - Include details about parameters, return values, and any possible exceptions or errors.

### **Task Checklist:**
- [ ] Implement the `zks_getBytecodeByHash` method.
- [ ] Write unit tests for the `zks_getBytecodeByHash` method.
- [ ] Add inline documentation for the `zks_getBytecodeByHash` method.

### **Additional Notes:**
If you're an external contributor wanting to work on this issue, please comment below to notify us. If you have questions about the requirements or need any assistance related to the implementation, don't hesitate to ask!

### PR DESCRIPTION
**Pull request by nbaztec**
# What :computer: 
* adds `zks_getBytecodeByHash`

# Why :hand:
* Allows developers to fetch deployed bytecodes via their hashes for investigation and debugging.

# Evidence :camera:
image
image

# Notes :memo:
* Fixes #144

### CODE REVIEW

File: e2e-tests/test/zks-apis.test.ts [hunk 1]
```diff
@@ -74,3 +74,25 @@ describe("zks_getTransactionDetails", function () {
...
+    const wallet = new Wallet(RichAccounts[0].PrivateKey);
+    const deployer = new Deployer(hre, wallet);
+    const artifact = await deployer.loadArtifact("Greeter");
+    const greeter = await deployContract(deployer, "Greeter", ["Hi"]);
+    expect(await greeter.greet()).to.eq("Hi");
+    // get the bytecode hash from the KnownCodesStorage log
+    const logs = await provider.send("eth_getLogs", [{ address: "0x0000000000000000000000000000000000008004" }]);
+    expect(logs).to.not.be.empty;
+    expect(logs[logs.length - 1].topics).to.have.lengthOf(3);
+    const bytecodeHash = logs[0].topics[1];
```
Comment (MexicanAce) [inline e2e-tests/test/zks-apis.test.ts:90]: I believe the topics are not in order... so unsure how to validate consistently. Especially when previous transactions have occurred 😞

Comment (nbaztec) [reply e2e-tests/test/zks-apis.test.ts]: I adapted the test to only use the contract's deployed block to fetch the logs from. The topics are always in order. The first topic is the solidity's event's keccak hash and the subsequent ones are the parameters.

**Review by MexicanAce — APPROVED** [review]
LGTM
